python/main.py

```diff
@@ -1,63 +1,400 @@
+        preds_mask = (
+            (b_input_ids != cls_tok)
+            & (b_input_ids != pad_tok)
+            & (b_input_ids != sep_tok)
+        )
 
-    dataset_filenames = ['train_combined.txt', 'dev_combined.txt', 'test_combined.txt']
-    dataset_file_list = [combined_path + fn for fn in dataset_filenames]
+        logits = logits.detach().cpu().numpy()
+        label_ids = torch.masked_select(b_labels, (preds_mask == 1))
+        val_batch_preds = np.argmax(logits[preds_mask.squeeze()], axis=1)
+        val_batch_labels = label_ids.to("cpu").numpy()
+        predictions.extend(val_batch_preds)
+        true_labels.extend(val_batch_labels)
 
-    # Training set
-    create_combined_en_dataset([conll_path + 'train.txt',
-                                ee_path + 'wnut17train.conll'],
-                                combined_path + 'train_combined.txt')
+        tmp_eval_accuracy = flat_accuracy(val_batch_labels, val_batch_preds)
 
-    # Validation set
-    create_combined_en_dataset([conll_path + 'valid.txt',
-                                ee_path + 'emerging.dev.conll'],
-                                combined_path + 'dev_combined.txt')
+        eval_loss += tmp_eval_loss.mean().item()
+        eval_accuracy += tmp_eval_accuracy
 
-    # Test set
-    create_combined_en_dataset([conll_path + 'test.txt',
-                                ee_path + 'emerging.test.annotated'],
-                                combined_path + 'test_combined.txt')
+        nb_eval_examples += b_input_ids.size(0)
+        nb_eval_steps += 1
 
-    # Standardize the labels on all 3 combined datasets
-    standardize_labels_and_save(dataset_file_list)
+    # Evaluate loss, acc, conf. matrix, and class. report on devset
+    pred_tags = [idx2tag[i] for i in predictions]
+    valid_tags = [idx2tag[i] for i in true_labels]
+    cl_report = classification_report(valid_tags, pred_tags)
+    conf_mat = annot_confusion_matrix(valid_tags, pred_tags)
+    eval_loss = eval_loss / nb_eval_steps
+    eval_accuracy = eval_accuracy / nb_eval_steps
 
-    # TODO: Pandas FutureWarning about read_table deprecation
-    # See if this can be relatively easily swapped out with read_csv
+    # Report metrics
+    print(f"Validation loss: {eval_loss}")
+    print(f"Validation Accuracy: {eval_accuracy}")
+    print(f"Classification Report:\n {cl_report}")
+    if args.verbose:
+        print(f"Confusion Matrix:\n {conf_mat}")
 
-    ### Below this line will be the code to train or run the NER model
-    ### Also, might add an arg into the ArgParser about plotting/exporting
-    ### some stats about the dataset and labels
+    # Save model and optimizer state_dict following every epoch
+    # TODO: Make this more robust by creating a timestamped directory in 'models'
+    # for each new run of the script - could save logs here too if/when logging
+    # is implemented
+    save_path = f"../models/{THIS_RUN}/train_checkpoint_epoch_{epoch}.tar"
```

Comment:
The first run of this had good results, but error when saving here: 

Validation loss: 0.17572927347615233
Validation Accuracy: 0.9378481123991649
Classification Report:
            precision    recall  f1-score   support

      LOC       0.81      0.77      0.79      1895
     MISC       0.54      0.05      0.10       719
      PRO       0.12      0.64      0.20        72
      PER       0.88      0.92      0.90      1514
      EVT       0.48      0.17      0.25        72

micro avg       0.76      0.69      0.72      4272
macro avg       0.77      0.69      0.69      4272

Confusion Matrix:
        EVT LOC MISC O ORG PER PRO
EVT     [12  3  0 56  1  0  0]
LOC     [   0 3308    0  615  189   43   46]
MISC    [   0   30  111 1925   30   15   49]
O       [    13    325     14 106418    539    364    600]
ORG     [   0  202    1 2437 5707   86  357]
PER     [   0   61    0  117   42 6247   15]
PRO     [  0   0   0  24  31   0 284]
Traceback (most recent call last):
  File "main.py", line 398, in <module>
    save_path,
  File "/home/nlopatina/.local/lib/python3.6/site-packages/torch/serialization.py", line 224, in save
    return _with_file_like(f, "wb", lambda f: _save(obj, f, pickle_module, pickle_protocol))
  File "/home/nlopatina/.local/lib/python3.6/site-packages/torch/serialization.py", line 147, in _with_file_like
    f = open(f, mode)
FileNotFoundError: [Errno 2] No such file or directory: '../models/10.04.2019, 19.16.28/train_checkpoint_epoch_1.tar'

Reply:
Likely due to permissions as discussed during our check-in